### PR TITLE
{tools}[GCCcore/12.3.0] Meson v1.3.1, meson-python v0.15.0

### PR DESCRIPTION
**Pull request by jfgrimm**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by jfgrimm** [comment]
Test report by @jfgrimm
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
gpu07.viking2.yor.alces.network - Linux Rocky Linux 8.8, x86_64, AMD EPYC 7643 48-Core Processor, 2 x NVIDIA NVIDIA A40, 535.86.10, Python 3.6.8
See https://gist.github.com/jfgrimm/50cc64142deb5daa825217f4f09eb72b for a full test report.

**Comment by jfgrimm** [comment]
@boegelbot: please test @ jsc-zen3

**Comment by bedroge** [comment]
Test report by @bedroge
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
interactive1 - Linux Rocky Linux 8.9, x86_64, AMD EPYC-Milan Processor (zen2), Python 3.6.8
See https://gist.github.com/bedroge/fd9d74220fc2928ddd255d3d4c0846ce for a full test report.

File: test/easyconfigs/easyconfigs.py [hunk 1]
```diff
@@ -394,6 +394,18 @@ def check_dep_vars(self, gen, dep, dep_vars):
...
+        # a build dependency elsewhere
+        if dep == 'Meson' and len(dep_vars) > 1:
+            for key in list(dep_vars):
+                ecs = dep_vars[key]
+                # filter out Meson variants that are only used as a dependency for meson-python
+                if all(ec.startswith('meson-python-') for ec in ecs):
+                    dep_vars.pop(key)
+                # always retain at least one dep variant
+                    if len(dep_vars) == 1:
+                        break
```
Comment (bedroge) [inline test/easyconfigs/easyconfigs.py:407]: Not sure if the `if` statement should be inside the other `if` or not, but either the comment or the `if` statement doesn't have the correct indentation here.

Comment (jfgrimm) [reply test/easyconfigs/easyconfigs.py]: good spot
functionally, I don't think it matters whether it's indented or not, but probably makes more sense to not indent

**Review by bedroge — APPROVED** [review]
Lgtm

**Comment by emdrago** [comment]
Test report by @emdrago
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in total)
gput052 - Linux RHEL 8.7 (Ootpa), x86_64, Intel(R) Xeon(R) CPU E5-2630 v4 @ 2.20GHz, 1 x NVIDIA NVIDIA GeForce RTX 2080 Ti, 535.54.03, Python 3.9.13
See https://gist.github.com/emdrago/3729452d9411286fe8d15bd627eae272 for a full test report.

**Pull request merged**